### ISSUE TITLE
Is it possible to mark an attribute as read-only?

### ISSUE THREAD

**Issue by l0b0** [pynamodb/PynamoDB#911]
Copied from https://stackoverflow.com/q/66361105/96588:

I'm working on a model which implements a read-only attribute with a simple hack:

```python
from pynamodb.models import Model


class MyModel(Model):
    […]
    updated_at = UTCDateTimeAttribute()

    def save(self):
        self.updated_at = datetime.now(timezone.utc)
        return super().save()
```

With this hack the user needs to provide `updated_at` when creating an instance, even though the value will be ignored.

That immediate issue could be solved by `updated_at = UTCDateTimeAttribute(null=True)`, but now the model signals that `updated_at` might not be part of the response, which is not the case.

Basically, how can I signal to the user that they *cannot* provide a value for a field (ideally trying to set `updated_at` should result in an error), but that every instance of this model will have a value for this field?

**Comment by ikonst** [comment]
I would make it private:
```python
_updated_at = UTCDateTimeAttribute(attr_name='updated_at')
```

If you wish to add a read-only property, you can always do:
```
@property
def updated_at() -> datetime:
  return self._updated_at
```
